src/test/java/gradle/cucumber/Steps.java

```diff
@@ -7,19 +7,22 @@
 import java.io.IOException;
 import java.net.http.HttpResponse;
 
-public class SimpleGetSteps {
+public class Steps {
     private String port;
+    private Support server;
     private HttpResponse<String> response;
 
     @Given("The server is running on port {string}")
     public void theServerIsRunningOnPort(String port) {
         this.port = port;
-        new Thread(new Support(port)).start();
+        this.server = new Support(port);
+        new Thread(server).start();
     }
 
     @When("I send method {string} for {string} to host at the specified port")
     public void iSendToHostAtTheSpecifiedPort(String method, String location) throws IOException, InterruptedException {
         this.response = new Client().request(port, method, location);
+        server.close();
```

Comment:
I feel like it makes more sense to have `server.close()` as `cleanup` function or in your last `@Then` block. But Damon might have some more thoughts on it

Reply:
That makes sense. I will look into cleanup functions thank you!